src/main/java/com/google/api/gax/core/PropertiesProvider.java

```diff
@@ -68,22 +69,30 @@ public static String loadProperty(Class<?> loadedClass, String propertyFilePath,
       Properties properties = new Properties();
       properties.load(loadedClass.getResourceAsStream(propertyFilePath));
       return properties.getProperty(key);
+    } catch (NullPointerException e) {
+      fileNotFoundWarning(loadedClass, propertyFilePath);
```

Comment:
Please explicitly check for the file not found using `File.exists` instead of catching `NullPointerException`. We could cause NPE for other reasons. We should probably never, ever catch an NPE unless it is caused by circumstances completely outside our control.